modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java

```diff
@@ -197,20 +211,85 @@ private JSONObject _getSegmentsExperienceJSONObject(
 		).put(
 			"segmentsExperienceName",
 			segmentsExperience.getName(_themeDisplay.getLocale())
+		).put(
+			"statusLabel",
+			() -> {
+				String statusLabelKey = "inactive";
+
+				if (segmentsExperienceIsActive) {
+					statusLabelKey = "active";
+				}
+
+				return _language.get(_httpServletRequest, statusLabelKey);
+			}
 		).put(
 			"url",
 			HttpComponentsUtil.setParameter(
-				PortalUtil.getCurrentURL(_httpServletRequest),
+				_portal.getCurrentURL(_httpServletRequest),
 				"segmentsExperienceId",
 				segmentsExperience.getSegmentsExperienceId())
 		);
 	}
 
+	private String _getSelectedSegmentsExperienceName(
+		SegmentsExperience segmentsExperience) {
+
+		SegmentsExperience parentSegmentsExperience =
+			_getParentSegmentsExperience(segmentsExperience);
+
+		if ((segmentsExperience != null) &&
+			(parentSegmentsExperience != null)) {
+
+			segmentsExperience = parentSegmentsExperience;
+		}
+
+		if (segmentsExperience != null) {
+			return segmentsExperience.getName(_themeDisplay.getLocale());
+		}
+
+		return SegmentsEntryConstants.getDefaultSegmentsEntryName(
+			_themeDisplay.getLocale());
+	}
+
+	private boolean _isActive(
+		SegmentsExperience segmentsExperience,
+		List<SegmentsExperience> segmentsExperiences) {
+
+		for (SegmentsExperience curSegmentsExperience1 : segmentsExperiences) {
+			if ((curSegmentsExperience1.getSegmentsEntryId() ==
+					segmentsExperience.getSegmentsEntryId()) ||
+				(curSegmentsExperience1.getSegmentsEntryId() ==
+					SegmentsEntryConstants.ID_DEFAULT)) {
+
+				if (curSegmentsExperience1.getSegmentsExperienceId() ==
```

Comment:
Can we move this condition before 

(curSegmentsExperience1.getSegmentsEntryId() ==
					segmentsExperience.getSegmentsEntryId()) ||
				(curSegmentsExperience1.getSegmentsEntryId() ==
					SegmentsEntryConstants.ID_DEFAULT)
					
					

So if this is not true, then for sure the experience is not active?

I think that maybe it will be faster.

What do you think?

Reply:
The code will look something like:

if (curSegmentsExperience1.getSegmentsExperienceId() != segmentsExperience.getSegmentsExperienceId()) {
  continue;
}

if ((curSegmentsExperience1.getSegmentsEntryId() ==
					segmentsExperience.getSegmentsEntryId()) ||
				(curSegmentsExperience1.getSegmentsEntryId() ==
					SegmentsEntryConstants.ID_DEFAULT)) {
	
	return true;				
}